src/sqlfluff/rules/L057.py

```diff
@@ -54,56 +54,45 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         self.unquoted_identifiers_policy: str
         self.allow_space_in_identifier: bool
 
-        if (
-            context.segment.name == "naked_identifier"
-            and identifiers_policy_applicable(
-                self.unquoted_identifiers_policy, context.parent_stack
-            )
-            and not (
-                context.segment.raw.replace("_", "").isalnum()
-                or (
-                    self.allow_space_in_identifier
-                    and context.segment.raw.replace("_", "").replace(" ", "").isalnum()
-                )
-            )
-        ):
-            return LintResult(anchor=context.segment)
-
-        # Exit early if not a quoted_identifier
-        if not context.segment.name == "quoted_identifier":
+        # Exit early if not a single identifier.
+        if context.segment.name not in ("naked_identifier", "quoted_identifier"):
```

Comment:
Early exit for both of the single identifier types rather than just quoted_identifier.